evalml/data_checks/datetime_format_data_check.py

```diff
@@ -118,21 +151,52 @@ def validate(self, X, y):
             )
             return messages
 
+        is_increasing = pd.DatetimeIndex(datetime_values).is_monotonic_increasing
         if not inferred_freq:
             col_name = (
                 self.datetime_column
                 if self.datetime_column != "index"
                 else "either index"
             )
-            messages.append(
-                DataCheckError(
-                    message=f"No frequency could be detected in {col_name}, possibly due to uneven intervals.",
-                    data_check_name=self.name,
-                    message_code=DataCheckMessageCode.DATETIME_HAS_UNEVEN_INTERVALS,
-                ).to_dict()
-            )
-
-        if not (pd.DatetimeIndex(datetime_values).is_monotonic_increasing):
+            frequencies = []
+            missing = []
+            for i in range(len(datetime_values) - 2):
+                freq = pd.infer_freq(datetime_values[i : i + 3])
```

Comment:
The missing dates part is really tricky. I like this but I saw that it couldn't catch this one case:

<img width="838" alt="Screen Shot 2022-01-27 at 4 59 34 PM" src="https://user-images.githubusercontent.com/41651716/151454266-243956be-2257-4ebb-a478-5f51a0fc6917.png">

If we change line 180 to read `len(set(frequencies)) == 2`, then it does catch this case as having missing dates.

<img width="832" alt="Screen Shot 2022-01-27 at 5 01 46 PM" src="https://user-images.githubusercontent.com/41651716/151454410-d0a8b594-c248-4854-b67c-28b6ee32effd.png">

Would this algorithm work?

```
missing_dates = datetime_values.diff(1).isna().nunique() > 1
```

If there are no missing dates, the first order differences should be constant.

Reply:
Now that I think about it, we can also use `datetime_values.diff(1)` to determine if there are any duplicate dates. 

```
duplicate_dates = (datetime_values.diff(1) == pd.Timedelta(0)).any()
```

And we should probably filter out the rows equal to Timedelta(0) prior to computing the uniques.

Reply:
@freddyaboulton the case you highlighted is an interesting one, and plays into my uncertainty about where the line of "missing dates" vs "uneven interval" is. IMO, it makes sense for your example to have uneven intervals rather than missing dates, since there is that segment in the middle where there is a noticeable two day interval. 

Of course, I can see an argument for the other way, so I'm curious to hear what you think!

Reply:
Just discussed this with @dsherry during office hours. Looks like the best way forward may be to raise both errors in this case, instead of just one or the other. I'll update this PR to reflect that shortly!

Reply:
@eccabay That's a good point, I would say that:

- if more than one frequency (aside from None) is identified in the dates column, then it would be uneven interval
- if one frequency (aside from None) is identified, and then None has also been identified **once**, then it would indicate missing dates
- if one frequency (aside from None) is identified, and then None has also been identified **more than once non-consecutively**, then it would indicate uneven interval.

But I think raising both errors might be the more straightforward path as you said!